### DISCUSSION TITLE
Could not locate image file for terrain...

### DISCUSSION THREAD

**Discussion by DemonTraitor** [kubilus1/autoortho#326] [Q&A]
### Please check existing docs and issues first.

- [X] I have checked for existing answers
- [X] This is not a bug or other "Issue"
- [X] I have read the FAQ and documentation

### System Details

- OS: [e.g. Windows 11]
- XPlane Version: XP12 beta 6
- AutoOrtho Version 0.5.1-3
- Filesystem type (NTFS, exFAT, etc)

**Additional context**
Add any other context about the problem here. (scenery version if applicable, etc, logfile snippet, etc)

### question

This is my set-up:

Autoortho installed to **E:\X-Plane\Custom Scenery\autoortho-data**

I have created Symbolic Links in my Custom Scenery folder for:

- yAutoOrtho_Overlays    linking to **E:\X-Plane\Custom Scenery\autoortho-data\yAutoOrtho_Overlays**
- z_autoortho   linking to **E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho**

And...

- z_ao_asi
- z_ao_aus_pac
- z_ao_eur

All linking to their respective folders in "E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho\scenery

**This my XP error:**

Error, could not locate image file for terrain
Custom Scenery/z_ao_eur/terrain/84000_128704_BI18_sea_overlay.ter
Custom Scenery/z_ao_eur/terrain/
../textures/84000_128704_BI18.dds
(REN_terrain.cpp:206)

And then CTD (autoortho is running)

I am assuming I have done something wrong?

Thank you

[autoortho.log](https://github.com/kubilus1/autoortho/files/12388495/autoortho.log)

**Comment by justkrook** [comment]
Same here, no Idea what to do next

**Comment by MsgLedeni** [comment]
What to do next?
Where can I find 4.4 to revert. it works flawlessly

**Reply by justkrook** [reply]
https://github.com/kubilus1/autoortho/releases/download/0.4.4/autoortho_win.zip

**Comment by MsgLedeni** [comment]
Thank You very much
Robert

**Comment by sergiolamares** [comment]
> ### Please check existing docs and issues first.
> * [x]  I have checked for existing answers
> * [x]  This is not a bug or other "Issue"
> * [x]  I have read the FAQ and documentation
> 
> ### System Details
> * OS: [e.g. Windows 11]
> * XPlane Version: XP12 beta 6
> * AutoOrtho Version 0.5.1-3
> * Filesystem type (NTFS, exFAT, etc)
> 
> **Additional context** Add any other context about the problem here. (scenery version if applicable, etc, logfile snippet, etc)
> 
> ### question
> This is my set-up:
> 
> Autoortho installed to **E:\X-Plane\Custom Scenery\autoortho-data**
> 
> I have created Symbolic Links in my Custom Scenery folder for:
> 
> * yAutoOrtho_Overlays    linking to **E:\X-Plane\Custom Scenery\autoortho-data\yAutoOrtho_Overlays**
> * z_autoortho   linking to **E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho**
> 
> And...
> 
> * z_ao_asi
> * z_ao_aus_pac
> * z_ao_eur
> 
> All linking to their respective folders in "E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho\scenery
> 
> **This my XP error:**
> 
> Error, could not locate image file for terrain Custom Scenery/z_ao_eur/terrain/84000_128704_BI18_sea_overlay.ter Custom Scenery/z_ao_eur/terrain/ ../textures/84000_128704_BI18.dds (REN_terrain.cpp:206)
> 
> And then CTD (autoortho is running)
> 
> I am assuming I have done something wrong?
> 
> Thank you
> 
> [autoortho.log](https://github.com/kubilus1/autoortho/files/12388495/autoortho.log)

I had exactly the same problem and reverted several times, back and forth to 0.4.4 and 0.5.0 (currently 0.5.1)
you need to make sure autoortho service is running (click run on app config window) BEBORE staring X-Plane (not a flight but the sim itself) only then symlinks/junctions/mounts are created in costum scenery or whatever folder you have autoortho installed from. Olny then X-Plane can acesse to scenery packeges simlinked on costum scenery folder. After you close/exit autoortho every simlink is closed/deleted. You can see all happening in the status windows, showing all loged data, from start to end of autoortho session.
It took a whle for me to understand how autoortho 0.5.x works, different then 0.4.4.
hope it helps

**Reply by DemonTraitor** [reply]
Yes, I understand exactly how it all works, and the folders/links work fine, but XP cannot handle it. So weird. Probably go back to RAW or previous version.

**Comment by MsgLedeni** [comment]
..."After you close/exit autoortho every symlink is closed/deleted." 
Not in my case, when I close AO Symlinks persist
![Screenshot_1 jpg c2d00e712c92538234c1c5d91a257a3c](https://github.com/kubilus1/autoortho/assets/78414004/7af8ca0a-01c5-4fee-a9fd-7bf92b3794a9)

**Reply by DemonTraitor** [reply]
The links stay, but you cannot access them - well, you should not be able to...

**Comment by kubilus1** [comment]
>    z_autoortho linking to E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho
>
> And...
>
>    z_ao_asi
>    z_ao_aus_pac
>    z_ao_eur
>
> All linking to their respective folders in "E:\X-Plane\Custom Scenery\autoortho-data\z_autoortho\scenery


@DemonTraitor  This is incorrect.   When running the 'active' based on your setup, the mounted folder will be "E:\X-Plane\Custom Scenery\autoortho-data\z_ao_na" etc etc.  You'll want to symlink _that_ to X-Plane's custom scenery folder.

Consider that whatever folder you tell AutoOrtho is for 'Custom Scenery' is at the exact same level and view as X-Plane's true 'Custom Scenery' folder.

**Reply by DemonTraitor** [reply]
Hi. I am on 1.5.1-3 using Dokan

I have autoortho folders installed into external folder "E:\X-Plane\Custom Scenery\autoortho-data"

I have tried **Symlinks** and **Shortcuts** to the default X-Plane 12 Customer Scenery Folder pointing to:

yAutoOrtho_Overlays
z_ao_eur
z_autoortho

The folders work in Windows Explorer, that you can navigate to them whether shortcuts or symlinks, but XP crashes everytime not being able to load **z_ao_eur**

Just cannot get this to work...

I have exhausted my efforts lol

**Comment by kubilus1** [comment]
Did you try the suggestion to use WinFSP instead of Dokan?

**Reply by DemonTraitor** [reply]
Yes, but XP kept re-ordering the "z_ao_eur" shortcut, which screwed the scenery... :/ This only happens with WinFSP

**Reply by chezhj** [reply]
Same here: I just tested a small flight with WinFSP with raw fsp on True. Had to play with the custom_scenery.ini and make it read only after changing the order. Xplane will complain, but runs fine. Might #183  be the cause of this? Will test longer flight tomorrow.

**Reply by chezhj** [reply]
I'm would really like to know how to debug this problem.
Today winFSP, same setup that was working yesterday, failed *.ter file could not be opened.
Removed winfsp, installed dokan. Old version (4) looked like a success, but it seems I've been fooled before, because the z\_ao\_ links are not created. So careful inspection of xplane log, tells me only some mesh is used.
Now trying Dokan, version 6, no threading, 
Waited half an hour... still no scenery loaded...
